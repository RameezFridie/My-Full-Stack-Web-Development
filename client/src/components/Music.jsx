// Import react modules
import React from 'react'
import './Styles/Music.css'

// Class of music used ton interact with back end and App.js file
class Music extends React.Component{
    // Set initial state using a constructor
    constructor(){
        super()
        this.state ={
            data: [],
            musicInput: ''
        }
    }
    // Display music that have been searched for
    newSearch(){
        fetch('/music')
            .then(res => res.json())
            .then(data => this.setState({data}, () => console.log('info fetched...', data)))
    }
    // Params used to search for books
    musicSearch = async () => {
        let search = this.state.musicInput.split(' ').join('+')
        const getMusic = await fetch(`/music?search=${search}&type=${this.state.type}`)
        let res = await getMusic.json()
        this.setState({
            data: res
        })
    }
      // Function that that saves book data in a json file that then writes to a favorites page
    favoriteMusic= (i) => {
        let favPic = {
            id: i.trackId,
            artist: i.artistName,
            artwork: i.artworkUrl100,
            track: i.trackName,
            sample: i.previewUrl
        }
        // Method used to post to json file and thn display
        fetch('/favoritesMusic', {
            method: 'POST',
            headers : {
                "Content-Type" : "application/json"
            },
            body: JSON.stringify(favPic)
        })
        alert('Added to favorites')
    }
    render(){
        return (
            <div>
                <div className='wrap'>
                    <h1 className="Title">iTunes</h1>
                    <div className='search'>
                        <input className='searchTerm' type="text" onChange={(e) => this.setState({musicInput: e.target.value})}/>
                        <button className="searchButton" onClick={() => this.musicSearch()}>
                        <i className="fa fa-search">üîé</i>
                        </button>
                    </div>
                </div>

                <fieldset>
                    {/* Display information from json file to the web page or UI */}
                    {this.state.data.map(sort_data => <article key={sort_data.trackId}>
                    <div className="cont">
                        <h3>{sort_data.artistName}</h3>
                        <h3>{sort_data.trackName}</h3>
                    </div>
                    <img src={sort_data.artworkUrl100} alt={sort_data.trackId}/>
                    <br/>
                    <audio className="audio" controls>
                        <source src={sort_data.previewUrl}type='audio/mpeg'></source>
                    </audio>
                    <button onClick={() =>this.favoriteMusic(sort_data)}>‚≠ê</button>
                    <hr/>
                    </article>)}

                </fieldset>
            </div>
        )
    }
}

export default Music