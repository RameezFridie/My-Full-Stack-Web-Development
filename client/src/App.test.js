// Importing modules and files being tested
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Music from './components/Music'


it.todo("api"), done => {
  function callBack(data) {
    expect(data).toBe("data")
    done()
  }
  callBack()
  const div = document.createElement("div")
  ReactDOM.render( <Music/> , div)
  ReactDOM.unmountComponentAtNode(div)
}

it("Passed", () => {
  const component = renderer.create(<Music/>)
  let app = component.toJSON()
  expect(app).toMatchSnapshot()
})
