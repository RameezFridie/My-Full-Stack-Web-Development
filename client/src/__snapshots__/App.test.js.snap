// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passed 1`] = `
<div>
  <div
    className="wrap"
  >
    <h1
      className="Title"
    >
      iTunes
    </h1>
    <div
      className="search"
    >
      <input
        className="searchTerm"
        onChange={[Function]}
        type="text"
      />
      <button
        className="searchButton"
        onClick={[Function]}
      >
        <i
          className="fa fa-search"
        >
          ðŸ”Ž
        </i>
      </button>
    </div>
  </div>
  <fieldset />
</div>
`;
